### PR TITLE
GH exposed key rule

### PR DESCRIPTION
**Pull request by BatteryCandy**
### Background

If you've ever walked into an organization with lots of AWS accounts and no control you are likely to not always know what people are doing with static IAM keys.  When static keys are uploaded to GH, AWS detects these and emails the account root email address.  But that might not always be the team that should be tracking this activity for longer term remediation of usage patterns within your company. 

### Changes

This detection monitors AWS cloudtrail for AWS service placing a deny all policy on the IAM user.  Its the same policy every time and therefore a good indicator for doing cleanup in your accounts when accidents happen.  Additionally, this type of activity is good for tracking for pushing internal changes to account usage practices. 

### Testing

Example event log has been included for testing.

### CODE REVIEW

File: aws_cloudtrail_rules/aws_key_compromised.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+def rule(event):
+    # Capture PutUserPolicy
+    request = event.get('requestParameters')
+    return event.get('eventName').startswith(
+        'PutUserPolicy') and request.get('policyName').startswith('AWSExposedCredentialPolicy_DO_NOT_REMOVE')
+
+
+def dedup(event):
+    return event.get('userIdentity', {}).get('userName')
```
Comment (jacknagz) [inline aws_cloudtrail_rules/aws_key_compromised.py:9]:
```suggestion
    return event['userIdentity'].get('userName')
```

You can skip `.get()` for the required fields (bolded and [found here](https://docs.runpanther.io/supported-logs/aws#aws-cloudtrail))

File: aws_cloudtrail_rules/aws_key_compromised.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
+AnalysisType: rule
+Filename: aws_key_compromised.py
+RuleID: AWS.Key.Compromised
```
Comment (jacknagz) [inline aws_cloudtrail_rules/aws_key_compromised.yml:3]: Let's call this `AWS.IAM.AccessKeyCompromised`

File: aws_cloudtrail_rules/aws_key_compromised.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+def rule(event):
+    # Capture PutUserPolicy
+    request = event.get('requestParameters')
+    return event.get('eventName').startswith(
```
Comment (jacknagz) [inline aws_cloudtrail_rules/aws_key_compromised.py:4]: I'm not seeing any other events that could match `PutRolePolicy` ([ref](https://docs.aws.amazon.com/IAM/latest/UserGuide/list_identityandaccessmanagement.html)), should we just do an equality check?

File: aws_cloudtrail_rules/aws_key_compromised.py [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+        'PutUserPolicy') and request.get('policyName').startswith('AWSExposedCredentialPolicy_DO_NOT_REMOVE')
+
+
+def dedup(event):
+    return event.get('userIdentity', {}).get('userName')
+
+
+def title(event):
+    user_identity = event.get('userIdentity', {})
```
Comment (jacknagz) [inline aws_cloudtrail_rules/aws_key_compromised.py:13]: This is also a required field, so direct access is safe!

File: aws_cloudtrail_rules/aws_key_compromised.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
+def rule(event):
+    # Capture PutUserPolicy
+    request = event.get('requestParameters')
+    return event.get('eventName').startswith(
+        'PutUserPolicy') and request.get('policyName').startswith('AWSExposedCredentialPolicy_DO_NOT_REMOVE')
```
Comment (jacknagz) [inline aws_cloudtrail_rules/aws_key_compromised.py:5]: Similar comment here on startswith(), should we just check that the string `AWSExposedCredentialPolicy` is in the `policyName` somewhere?

File: aws_cloudtrail_rules/aws_key_compromised.py (hunk 1 above)
Comment (BatteryCandy) [reply aws_cloudtrail_rules/aws_key_compromised.py]: 👍

File: aws_cloudtrail_rules/aws_key_compromised.yml (hunk 2 above)
Comment (BatteryCandy) [reply aws_cloudtrail_rules/aws_key_compromised.yml]: 👍

File: aws_cloudtrail_rules/aws_key_compromised.py (hunk 5 above)
Comment (BatteryCandy) [reply aws_cloudtrail_rules/aws_key_compromised.py]: made this an equality check as this is always the policy thats applied

**Comment by BatteryCandy** [comment]
I think I have everything addressed now.  Thanks for the review and let me know if anything you'd like resolved.

**Comment by austinbyers** [comment]
Merging because tests pass and there is an approval. Thanks @BatteryCandy !

**Pull request merged**